core/drivers/stm32_rtc.c

```diff
@@ -321,11 +321,11 @@ static unsigned long long stm32_rtc_diff_date_ms(struct optee_rtc_time *current,
 				ref->tm_mday + current->tm_mday;
```

Comment:
Maybe we can use rtc_timcmp to ensure ref is prior to current (also in stm32_rtc_diff_time_ms).
Otherwise the implementation looks good to me.